### ISSUE TITLE
"View all" places returns fewer results than expected after a search

### ISSUE THREAD

**Issue by MomoMoses** [johnjung/bmrcportal#3]
To replicate the issue, start with a search on "Chicago State" (with quotes). In the Places facet,

**Comment by MomoMoses** [comment]
Also, it is perplexing to see that most of the place names in the "View all places" modal have only (1) hit. This doesn't feel right in terms of the data in an intuitive sense.

**Comment by johnjung** [comment]
This seems to be working the way I understood the spec. In the example search above, when the user searches for "Chicago State", the facets that appear in the sidebar change- the are always applicable to the current search term and any other facets that are currently active. (In other words, they always filter the current search, facets do not apply to the entire corpus when a search term or other facet(s) are currently active.) So if you were to click "view all places" and select "Englewood" on its own, right you get you 8 results, but if you do that after executing a search for "Chicago State" you get a single result.

Regarding the number of places with a single result- if you just go to an empty search result page (just visit /search/, without any query) you can see places in the sidebar that occur more than once in the corpus. If I select "view all places" and scroll to the bottom, I will find a long list of places that occur only once. I tested several of these by grepping through the data:

$ grep -r West\ Lawn\ \(Chicago,\ Ill\. *
chm/BMRC.CHM.SCC.xml:            <geogname source="lcsh">West Lawn (Chicago, Ill.)</geogname>

$ grep -r Anchorage-Frankfurt *
nu/BMRC.NU.TRANSP-LIB_MENUS.xml:            <unittitle encodinganalog="245" label="Title">Menu: Tokyo, Japan to Anchorage-Frankfurt,
nu/BMRC.NU.TRANSP-LIB_MENUS.xml:                <geogname encodinganalog="651" source="lcnaf">Anchorage-Frankfurt,

$ grep -r Deerfield\ \(Ill\. *
lake/BMRC.LAKE.UNTERMYER.xml:            <geogname source="lcsh">Deerfield (Ill.)</geogname>

Please let me know if you find specific examples of places or other facets where the count in the sidebar is inaccurate. If so, lets open a new issue to see what's going on there.

**Comment by MomoMoses** [comment]
OK. This could be a project for Phase 2 to enrich the data such that umbrella terms are used for some oft-used or very significant terms. Example: "Chicago neighborhoods" -->Englewood or "European cities" --> Frankfurt

**Comment by MomoMoses** [comment]
Perhaps explanatory text in the UI is in order: "View all related Topics" not "View all topics". The latter would be reserved for a Browse function.

**Comment by johnjung** [comment]
Please confirm that you'd like me to make this change- otherwise I'll close the issue out.

**Comment by MomoMoses** [comment]
Let's make the change and then I'll see what reactions I get from users. Also, whether the UI element can accommodate all those characters or not! Question for Kathy. But let's see what it looks like.

**Comment by johnjung** [comment]
This is now changed in https://github.com/johnjung/bmrcportal/commit/94e7bf0c3917f664a73b9d93ebac9263dcf24539, and live on bmrcportal-test.lib. Please take a look- if this looks ok we can close this issue out.

**Comment by MomoMoses** [comment]
Oh heck. That looks weird when you say "View all related people"! How about this "View all relevant people"...?

**Comment by johnjung** [comment]
This is updated inhttps://github.com/johnjung/bmrcportal/commit/abe7a896d158c93f421985cf5fb34a720645d53e, I'll push this change to bmrcportal-test.lib soon.

**Comment by johnjung** [comment]
This change is now live on bmrcportal-test.lib.

**Comment by MomoMoses** [comment]
Great! Another UI/UX question: should the heading at the top of the modal change? "View All Relevant [facet]"?

**Comment by johnjung** [comment]
The current text on the popup is "VIEWING ALL TOPICS". Please confirm: would you like me to change this to "VIEWING ALL RELEVANT <FACET>"?

**Comment by MomoMoses** [comment]
No it would be "Viewing All Relevant Topics" for example, or whatever the
facet name is.

**Comment by johnjung** [comment]
Thanks for confirming. (That is actually what I meant to type.) This is updated in https://github.com/johnjung/bmrcportal/commit/552ef109c651a3cd42f714fff6ae93959ff0ee8d, and this change is now live on the site. Please let me know if this looks ok.

**Comment by MomoMoses** [comment]
Looks fine! May wish to weigh in with Kathy to see if this phrasing and look is OK.
